src/tools/test-targets.ts

```diff
@@ -39,3 +42,154 @@ export const listTestTargets = async (options: ListOptions): Promise<void> => {
     );
   }
 };
+
+export type TestTargetPullData = components["schemas"]["TestTargetPullData"];
+
+type TestCases = components["schemas"]["TestTargetPullData"]["testCases"];
+
+const writeYaml = (data: TestTargetPullData, destination?: string) => {
+
+  for(const testCase of data.testCases) {
+    const folderName = buildFolderName(testCase, data.testCases, destination);
+    const testCaseFilename = buildFilename(testCase, folderName);
+    // Ensure folder exists
+    fs.mkdirSync(folderName, { recursive: true });
+    fs.writeFileSync(path.join(folderName, testCaseFilename), yaml.stringify(testCase));
+  }
+
+  //fs.writeFileSync("test-target.yaml", yaml.stringify(data));
+};
+
+export const buildFolderName = (testCase: TestCases[number], testCases: TestCases, destination?: string): string => {
+  type T = { id?: string | number; prerequisiteId?: string | number; description?: string };
+
+  const toSegment = (desc: string | undefined): string => {
+    const raw = String(desc ?? "root");
+    const normalized = raw
+      .normalize("NFKD")
+      .replace(/[\u0300-\u036f]/g, "");
+    const tokens = normalized
+      .split(/[^a-zA-Z0-9]+/)
+      .filter(Boolean);
+    const camel = tokens
+      .map((t, i) => {
+        const lower = t.toLowerCase();
+        if (i === 0) return lower;
+        return lower.charAt(0).toUpperCase() + lower.slice(1);
+      })
+      .join("");
+    return camel || "root";
+  };
+
+  const idOf = (tc: TestCases[number]) => String((tc as T).id ?? "");
+  const prereqOf = (tc: TestCases[number]) =>
+    (tc as T).prerequisiteId === undefined || (tc as T).prerequisiteId === null
+      ? undefined
+      : String((tc as T).prerequisiteId);
+
+  const findById = (id: string): TestCases[number] | undefined =>
+    testCases.find((tc) => idOf(tc) === id);
+
+  const segments: string[] = [];
+  const visited = new Set<string>();
+
+  let currentPrereq = prereqOf(testCase);
+  while (currentPrereq !== undefined) {
+    if (visited.has(currentPrereq)) {
+      throw new Error(`Cycle detected in prerequisites at id ${currentPrereq}`);
+    }
+    visited.add(currentPrereq);
+
+    const prereqTc = findById(currentPrereq);
+    if (!prereqTc) {
+      throw new Error(`Prerequisite with id ${currentPrereq} not found in test cases`);
+    }
+
+    const seg = toSegment((prereqTc as T).description);
```

Comment:
I would really like to get rid of all the casts, I feel like they're not necessary? 🤔